app/controllers/admin/email_controller.rb

```diff
@@ -69,6 +69,19 @@ def send_digest
     end
   end
 
+  def smtp_should_reject
+    params.require(:from)
+    params.require(:to)
+    # These strings aren't localized; they are sent to an anonymous SMTP user.
+    if User.find_by_email(params[:from]).nil? && !SiteSetting.enable_staged_users
+        render json: { reject: true, reason: "Mail from your address is not accepted. Do you have an account here?" }
+    elsif Email::Receiver.new(params[:from]).check_address(params[:to]).nil?
```

Comment:
I think it would be better to promote the `check_address` method to a class method.

`Email::Receiver.check_address(Email.downcase(params[:to])).nil?` is shorter